src/Client/Handlers/DelegateNotificationHandler.cs

```diff
@@ -50,7 +52,7 @@ public DelegateNotificationHandler(string method, NotificationHandler<TNotificat
             await Task.Yield();
 
             Handler(
-                notification != null ? notification.ToObject<TNotification>() : default(TNotification)
+                notification != null ? notification.ToObject<TNotification>(Serializer.Instance.JsonSerializer /* Fix me: this is ugly */) : default(TNotification)
```

Comment:
Hmm, I'm thinking maybe we could make the the `JsonSerializerSettings` assignable via a property on `ICustomizeJsonSerializer` (new interface with a `SerializerSettings` property that `INotificationHandler` and friends could inherit from)? Either that or pass it into the `Invoke()` methods but the property seems cleaner.

Reply:
What if we change `Invoke` to take an `object` instead and deserialize the value before we call `Invoke`?

Reply:
Yeah that would be simpler - nice one :)

Reply:
Mind you, how do you know which type to deserialise into? I think that's why I did it in the handler because the parameter type is known.

Reply:
I suppose the handler could expose a `ParametersType` / `PayloadType` property?